engine/common/synchronization/engine_test.go

```diff
@@ -2,187 +2,26 @@ package synchronization
 
 import (
 	"context"
-	"fmt"
-	"io"
 	"math"
 	"testing"
 	"time"
 
-	"github.com/rs/zerolog"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
-	"github.com/stretchr/testify/suite"
 
-	mockconsensus "github.com/onflow/flow-go/engine/consensus/mock"
-	"github.com/onflow/flow-go/model/chainsync"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/model/flow/filter"
 	"github.com/onflow/flow-go/model/messages"
 	synccore "github.com/onflow/flow-go/module/chainsync"
-	"github.com/onflow/flow-go/module/id"
 	"github.com/onflow/flow-go/module/irrecoverable"
 	"github.com/onflow/flow-go/module/metrics"
-	module "github.com/onflow/flow-go/module/mock"
 	netint "github.com/onflow/flow-go/network"
 	"github.com/onflow/flow-go/network/channels"
-	"github.com/onflow/flow-go/network/mocknetwork"
-	"github.com/onflow/flow-go/network/p2p/cache"
-	protocolint "github.com/onflow/flow-go/state/protocol"
-	protocolEvents "github.com/onflow/flow-go/state/protocol/events"
-	protocol "github.com/onflow/flow-go/state/protocol/mock"
-	storerr "github.com/onflow/flow-go/storage"
-	storage "github.com/onflow/flow-go/storage/mock"
 	"github.com/onflow/flow-go/utils/rand"
 	"github.com/onflow/flow-go/utils/unittest"
 )
 
-func TestSyncEngine(t *testing.T) {
-	suite.Run(t, new(SyncSuite))
-}
-
-type SyncSuite struct {
```

Comment:
Extract test suite code into separate file (`engine/common/synchronization/engine_suite_test.go`) since it's used by 2 test files (`engine/common/synchronization/engine_spam_test.go` and `engine/common/synchronization/engine_test.go`)